crates/matrix-sdk-crypto/src/verification/machine.rs

```diff
@@ -389,18 +389,22 @@ impl VerificationMachine {
             }
             AnyVerificationContent::Cancel(c) => {
                 if let Some(verification) = self.get_request(event.sender(), flow_id.as_str()) {
-                    verification.receive_cancel(event.sender(), c);
+                    if !verification.is_cancelled() {
+                        verification.receive_cancel(event.sender(), c);
+                    }
                 }
 
                 if let Some(verification) = self.get_verification(event.sender(), flow_id.as_str())
                 {
-                    match verification {
-                        Verification::SasV1(sas) => {
-                            // This won't produce an outgoing content
-                            let _ = sas.receive_any_event(event.sender(), &content);
+                    if !verification.is_cancelled() {
+                        match verification {
+                            Verification::SasV1(sas) => {
+                                // This won't produce an outgoing content
+                                let _ = sas.receive_any_event(event.sender(), &content);
+                            }
+                            #[cfg(feature = "qrcode")]
+                            Verification::QrV1(qr) => qr.receive_cancel(event.sender(), c),
```

Comment:
What is this about? This doesn't seem to be directly related to the PR?

Reply:
Oh damn, these were part of some experiments I did to check whether the issues could be fixed this way and I was *sure* I had reverted them. Apparently not, sorry.

Reply:
Removed in https://github.com/matrix-org/matrix-rust-sdk/pull/4705/commits/deaaca1aef4448add7ed1cce54db35bb97c6b610